src/data_info/services/stat.clj

```diff
@@ -27,6 +28,24 @@
   [stat-map]
   (= (:permission stat-map) :own))
 
+(defn- needs-key?
+  [included-keys needed-key?]
+  (if (contains? included-keys needed-key?)
+    true
+    (cond
+      (= needed-key? :permission)
+      (contains? included-keys :share-count)
+
+      (= needed-key? :type)
+      (not (empty? (cset/intersection #{:infoType :content-type :file-count :dir-count} included-keys)))
```

Comment:
Hm, I don't like `required-keys`, but maybe something like `requested-keys`? I'd called it `included-keys` because it's the keys which remain after processing the include and exclude lists into just includes. You're right that using a `or` and `case` is nicer for sure though!